packages/react-hooks/src/hooks/use-cart/use-cart.provider.tsx

```diff
@@ -28,15 +28,35 @@ export type CartActions = {
   clearCart: () => void;
 };
 
-export type CartProviderProps = {};
+export type CartProviderProps = {
+  useLocalStorage: boolean;
+  children: [ReactNode];
+};
 export type CartContextValue = null | CartState;
 export type CartActionContextValue = null | CartActions;
 
 const CartContext = React.createContext<CartContextValue>(null);
 const CartActionContext = React.createContext<CartActionContextValue>(null);
 
-export const CartProvider: FC<CartProviderProps> = ({ children }) => {
-  const [state, dispatch] = useReducer(cartReducer, initialState);
+export const CartProvider: FC<CartProviderProps> = ({ useLocalStorage, children }) => {
+  let cart: CartItem[] = []
+  let setCacheItem: Function = () => { }
```

Comment:
Rather than initialize this value inside the `CartProvider`, how do you feel about doing it in the `reducer` instead? I'm thinking something like:

```js
...
	const cartReducer = (
  state: CartState,	  state: CartState,
  action: CartReducerAction	  action: CartReducerAction
): CartState => {	): CartState => {
  const setCacheItem = state.useLocalStorage ? window.localStorage.setItem : () => {};
  
  switch (action.type) {	  switch (action.type) { ... }
```

Inside `CartProvider`, we'd only need to do something like:

```
const hasWindow = typeof window !== 'undefined';

const cart = useLocalStorage && hasWindow
  ? JSON.parse(window.localStorage.getItem('cart'))
  : [];

  const [state, dispatch] = useReducer(
    cartReducer,
    {
      ...initialState,
      cart,
      useLocalStorage: hasWindow && useLocalStorage
    }
  );

```

I think this keeps the code a bit cleaner to maintain, further separates concerns and removes the need to `bind` `setItem`. Thoughts?

Reply:
I think this sounds good. Still seems we need to bind setItem to localStorage.